File: Sem2/PO_project/python/animals/turtle.py
from __future__ import annotations

from python.organism import Animal


class Turtle(Animal):
    def __init__(self, strength: int, initiative: int, x: int, y: int, game) -> None:
        super().__init__(strength, initiative, x, y, game)

    def get_color(self) -> tuple[int, int, int]:
        return (0, 128, 0)

    def can_deflect(self, attacker) -> bool:
        return attacker.get_strength() < 5

    def get_organism(self):
        return self

    def reproduce(self, x: int, y: int):
        return Turtle(2, 1, x, y, self.game)

File: Sem2/PO_project/Java/animals/Turtle.java
package Java.animals;
import Java.Animal;
import Java.Game;
import Java.Organism;
import java.awt.Color;

public class Turtle extends Animal {
    public Turtle(int strength, int initiative, int x, int y, Game game) {
        super(strength, initiative, x, y, game);
    }

    @Override
    public Color getColor() {
        return new Color(0, 128, 0); // green
    }

    @Override
    public boolean canDeflect(Organism attacker) {
        if (attacker.getStrength() < 5) {
            return true;
        }
        return false;
    }

    @Override
    public Organism getOrganism() {
        return this;
    }

    @Override
    protected Animal reproduce(int x, int y) {
        return new Turtle(2, 1, x, y, game);
    }
    
}
